cmd/hook.go

```diff
@@ -101,6 +102,37 @@ Gitea or set your environment appropriately.`, "")
 	total := 0
 	lastline := 0
 
+	commsChan := make(chan string, 10)
+	tickerChan := make(chan struct{})
+	go func() {
+		sb := strings.Builder{}
+		hasWritten := false
+		ticker := time.NewTicker(1 * time.Second)
+	loop:
+		for {
+			select {
+			case s, ok := <-commsChan:
+				if ok {
+					sb.WriteString(s)
```

Comment:
```suggestion
					if hasWritten {
						os.Stdout.WriteString(s)
						os.Stdout.Sync()
					} else {
						sb.WriteString(s)
					}
```

Then the next block (`else`) should consist of only `break loop`.